src/coreclr/jit/assertionprop.cpp

```diff
@@ -1242,7 +1242,7 @@ AssertionIndex Compiler::optCreateAssertion(GenTree* op1, GenTree* op2, optAsser
                     if (op2->gtOper == GT_CNS_INT)
                     {
                         ssize_t iconVal = op2->AsIntCon()->IconValue();
-                        if (varTypeIsSmall(lclVar) && op1->OperIs(GT_STORE_LCL_VAR))
+                        if (varTypeIsSmall(lclVar))
```

Comment:
Do we risk creating a wrong complementary assertion by doing this? E.g. if we have

```csharp
ushort foo = 123;
if (foo == 70000)
{
  // assertion saying that "foo == 70000 & 0xffff"; in reality this block is unreachable
}
else
{
  // assertion saying that "foo != 70000 & 0xffff"; but that's false, we will get here even with foo == 70000 & 0xffff
}
```

I think we should either delete the `FitsIn` assert or avoid creating the assertion in the unreachable case.